molecule/default/tasks/rollback.yml

```diff
@@ -77,20 +67,72 @@
                     ports:
                       - containerPort: 80
       ignore_errors: yes
-      register: output
+      register: crash
 
     - name: Rolling Back the crashed deployment
-      k8s_rollback:
-        api_version: apps/v1
-        kind: Deployment
-        name: nginx-deploy
-        namespace: "{{ namespace }}"
-      when: output.failed
-      register: output
-
-    - name: Show output
-      debug:
-        var: output
+      block:
+        - name: Read the deployment
+          k8s_info:
+            kind: Deployment
+            name: nginx-deploy
+            namespace: "{{ namespace }}"
+          register: deployment
+
+        - set_fact:
+            failed_version: "{{ deployment.resources[0].metadata.annotations['deployment.kubernetes.io/revision'] }}"
+
+        - name: Rolling Back the crashed deployment (check mode)
+          k8s_rollback:
+            api_version: apps/v1
+            kind: Deployment
+            name: nginx-deploy
+            namespace: "{{ namespace }}"
+          register: result
+          check_mode: yes
+
+        - assert:
+            that:
+              - result is changed
+
+        - name: Read the deployment
+          k8s_info:
+            kind: Deployment
+            name: nginx-deploy
+            namespace: "{{ namespace }}"
+          register: deployment
+
+        - name: Validate that Rollback using check_mode did not changed the Deployment
+          assert:
+            that:
+              - failed_version == deployment.resources[0].metadata.annotations['deployment.kubernetes.io/revision']
+
+        - name: Rolling Back the crashed deployment
+          k8s_rollback:
+            api_version: apps/v1
+            kind: Deployment
+            name: nginx-deploy
+            namespace: "{{ namespace }}"
+          register: result
+
+        - name: assert rollback is changed
+          assert:
+            that:
+              - result is changed
+
+        - name: Read the deployment once again
+          k8s_info:
+            kind: Deployment
+            name: nginx-deploy
+            namespace: "{{ namespace }}"
+          register: deployment
+
+        - name: Validate that Rollback using check_mode did not changed the Deployment
```

Comment:
But the previous task is no longer performed with check_mode https://github.com/ansible-collections/kubernetes.core/pull/255/files#diff-f71da7f1f86c5838171a1e730fbf7185ceae88d529d4455cf26c74b069291e18R109. I would probably expect to have a change. For this reason I thought the name of the tasks should probably be rephrased. But maybe I am wrong.

Reply:
ok you are right